// @polkadot/metadata/decorate/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.expandMetadata = expandMetadata;
Object.defineProperty(exports, "constantsFromMeta", {
  enumerable: true,
  get: function () {
    return _fromMetadata.constantsFromMeta;
  }
});
Object.defineProperty(exports, "extrinsicsFromMeta", {
  enumerable: true,
  get: function () {
    return _fromMetadata2.extrinsicsFromMeta;
  }
});
Object.defineProperty(exports, "storageFromMeta", {
  enumerable: true,
  get: function () {
    return _fromMetadata3.storageFromMeta;
  }
});

var _util = require("@polkadot/util");

var _Metadata = require("../Metadata");

var _fromMetadata = require("./consts/fromMetadata");

var _fromMetadata2 = require("./extrinsics/fromMetadata");

var _fromMetadata3 = require("./storage/fromMetadata");

// Copyright 2017-2020 @polkadot/metadata authors & contributors
// SPDX-License-Identifier: Apache-2.0

/**
 * Expands the metadata by decoration int consts, query and tx sections
 */
function expandMetadata(registry, value) {
  (0, _util.assert)(value instanceof _Metadata.Metadata, 'You need to pass a valid Metadata instance to Decorated');
  return {
    consts: (0, _fromMetadata.constantsFromMeta)(registry, value),
    query: (0, _fromMetadata3.storageFromMeta)(registry, value),
    tx: (0, _fromMetadata2.extrinsicsFromMeta)(registry, value)
  };
}
